### PR TITLE
Ethconnect Support w/ Smart Contract Registration

### PR DESCRIPTION
**Pull request by drewmarshburn**
Adds support for deploying ethconnect alongside Firefly. 

TODO:
- [x] @hfuss Please see my question about splitting the persistent volumes.
- [x] Needs to be tested.

### CODE REVIEW

File: charts/firefly/values.yaml [hunk 1]
```diff
@@ -432,3 +432,94 @@ erc20:
...
+    #   cpu: 100m
+  #   memory: 128Mi
+
+  nodeSelector: {}
+
+  tolerations: []
+
+  affinity: {}
+
+  # TODO: @Hayden should I just have one big volume here at /ethconnect?
```
Comment (drewmarshburn) [inline charts/firefly/values.yaml:510]: @hfuss

Comment (hfuss) [reply charts/firefly/values.yaml:510]: yep think we should do what we did with the DX `peers` volume and use subPaths to split it into two folders. Will do this as part of my testing and fixing the linting.

File: charts/firefly/templates/_helpers.tpl [hunk 2]
```diff
@@ -164,7 +177,11 @@ blockchain:
   type: ethereum
   ethereum:
     ethconnect:
+      {{ if .Values.ethconnect.enabled }}
```
Comment (hfuss) [inline charts/firefly/templates/_helpers.tpl:180]: nice thanks for adding this, we'll need to add the same thing to the ERC1155 and ERC20 env vars as well.

**Comment by hfuss** [comment]
Since I ended up contributing a good bit to test this, @drewmarshburn mind self-reviewing? And then a review from @calbritt and / or @peterbroadhurst would great before we merge this and release `v0.1.0` including #13

**Review by drewmarshburn — COMMENTED** [review]
Re-reviewed and approved (Github won't let me formally approve as the author).

**Review by hfuss — APPROVED** [review]

**Pull request merged**